includes/classes/shopping_cart.php

```diff
@@ -786,7 +786,8 @@ function calculate() {
                 }
               if ($attribute_price->fields['attributes_discounted'] == '1') {
                 // calculate proper discount for attributes
-                $new_attributes_price = zen_get_discount_calc($product->fields['products_id'], $attribute_price->fields['products_attributes_id'], $attribute_price->fields['options_values_price'], $qty);
+                $new_attributes_price = zen_get_discount_calc($product->fields['products_id'], $attribute_price->fields['products_attributes_id'], /*$attribute_price->fields['options_values_price']*/ $attribute_price->fields["options_values_price"] + (zen_get_products_price_is_priced_by_attributes($attribute_price->fields["products_id"]) ? zen_products_lookup($attribute_price->fields["products_id"], 'products_price') : 0), $qty);
```

Comment:
I "get" that you've commented-out the "old" and then added new, but to comment-out the exact same variable and then add it back in again is redundantly redundant.

```diff
-$attribute_price->fields["options_values_price"]
+/*$attribute_price->fields['options_values_price']*/ $attribute_price->fields["options_values_price"] + foo...
```

Same on line 1019